### PR TITLE
Add release-4.15 builds

### PR DESCRIPTION
**Pull request by LorbusChris**
Also remove okd-coreos-build pipeline

The okd-coreos-all pipeline can now be configured via parameters, letting us use that for development builds without having to maintain a separate pipeline definition.

Tested locally

### CODE REVIEW

File: environments/kind/pipelineruns/okd-coreos-4.14-dev-pipelinerun.yaml
Comment (jcapiitao) [inline environments/kind/pipelineruns/okd-coreos-4.14-dev-pipelinerun.yaml:1]: IIUC `release-stream` value should be `stable`

Comment (joelcapitao) [inline environments/kind/pipelineruns/okd-coreos-4.14-dev-pipelinerun.yaml:1]: IIUC `release-stream` value should be `stable`

File: base/batch/cronjobs/okd-coreos-all-4.14-create-daily.yaml
Comment (joelcapitao) [inline base/batch/cronjobs/okd-coreos-all-4.14-create-daily.yaml:1]: `RELEASE_STREAM` value should be `stable`

**Review by joelcapitao — COMMENTED** [review]
Just a few in-line comments, other than that LGTM :+1:

File: environments/kind/pipelineruns/okd-coreos-4.14-dev-pipelinerun.yaml
Comment (LorbusChris) [reply environments/kind/pipelineruns/okd-coreos-4.14-dev-pipelinerun.yaml:1]: For now, 4.13 is still stable, while 4.14 and 4.15 go into next

Comment (LorbusChris) [reply environments/kind/pipelineruns/okd-coreos-4.14-dev-pipelinerun.yaml:1]: There's probably room for improvement for handling building 2 different `next` versions

**Comment by LorbusChris** [comment]
Everything is operational. Merging

**Pull request merged**